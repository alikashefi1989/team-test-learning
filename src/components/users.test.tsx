import { render, screen, fireEvent, waitForElementToBeRemoved, waitFor } from '@testing-library/react';
import Users from './users'
import { create, ReactTestRendererJSON } from 'react-test-renderer';
// import {screen} from '@testing-library/dom'
describe('users component', () => {

  const renderComponent = (props?: any) => create(<Users {...props} />);

  it('should render users component', () => {
    const component = renderComponent().toJSON();
    expect(component).toMatchSnapshot()
  });

  it('should render users cmp', async () => {
    const { getByText, getByTestId } = render(
      <Users />,
    )

    const button = getByTestId('custom-element')
    fireEvent.click(button);
    const loading = getByText('Loading...')
    expect(loading).toBeInTheDocument()
    expect(loading).toBeTruthy()

    const textData = await waitFor<HTMLElement>(() => getByText('data...'),{
      interval:2,
      timeout:10
    })
    console.log(textData,"jafar")
    expect(textData).toBeTruthy()
    expect(loading).toBeInTheDocument()


    // test('shows how async / await works', async () => {

    //   try {
    //     const value = await Promise.resolve(true);

    //   } catch (error) {

    //   }
    // });

  })
})


