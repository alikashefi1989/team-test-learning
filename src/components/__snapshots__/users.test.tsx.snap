// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users component should render users component 1`] = `
<button
  data-testid="load-button"
  onClick={[Function]}
>
  load
</button>
`;
